engines/native/system/user_linux.go

```diff
@@ -133,7 +164,30 @@ func CreateUser(homeFolder string, groups []*Group) (*User, error) {
 		return nil, fmt.Errorf("Failed to chown homeFolder, error: %s", err)
 	}
 
-	return &User{uint32(uid), uint32(gid), name, homeFolder}, nil
+	// Find group ids
+	osUser, err := user.Lookup(name)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to lookup user: %s", name)
+	}
+	sgids, err := osUser.GroupIds()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to list groups for current user")
+	}
+	gids := make([]uint32, len(sgids))
+	for i, sgid := range sgids {
+		igid, err := strconv.Atoi(sgid)
+		gids[i] = uint32(igid)
+		if err != nil {
+			panic(errors.Wrap(err, "Could not convert group id to integer"))
+		}
+	}
```

Comment:
This part of looking for a user's groups and transforming that to an array of integer gids should go in a internal utility function to avoid code duplication.

Reply:
good point... I should stop my copy/pasta code...